Cython/Compiler/ExprNodes.py

```diff
@@ -11007,7 +11005,11 @@ def get_type_info_type(self, env):
 
     cpp_message = 'typeid operator'
 
+    def type_dependencies(self, env):
```

Comment:
It looks like `TypeIdNode` (which was the other node to default to error_type) is a bit of a mess of fully analysing its operand. Thus it's pretty fragile to changes like this. Blocking `type_dependencies` seems to fix it, but it isn't a hugely satisfactory solution really.

Reply:
Looking again, I think the right thing to do is probably either to override `generate_evaluation_code` or to drop the `operand` attribute entirely after `analyse_types` since it's not needed so it's most robust to just avoid evaluating it completely. I'll mark this as draft for the moment and change it properly later...